Neos.Neos/Classes/AssetUsage/Service/GlobalAssetUsageService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\AssetUsage\Service;
+
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryServiceInterface;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Neos\AssetUsage\AssetUsageFinder;
+use Neos\Neos\AssetUsage\Dto\AssetUsageFilter;
+use Neos\Neos\AssetUsage\Dto\AssetUsages;
+use Neos\Flow\Annotations as Flow;
+use Neos\Neos\AssetUsage\Projection\AssetUsageRepository;
+
+/**
+ * @internal
+ */
+class GlobalAssetUsageService implements ContentRepositoryServiceInterface
+{
+    /**
+     * @var array<string, mixed>
+     */
+    #[Flow\InjectConfiguration(path: "AssetUsage")]
+    protected array $flowSettings;
+
+    /**
+     * @var array<string, ContentRepository>
+     */
+    protected ?array $repositories = null;
+
+    public function __construct(
+        protected readonly ContentRepositoryRegistry $contentRepositoryRegistry,
+    ) {
+    }
+
+    public function findAssetUsageByAssetId(string $assetId): AssetUsages
+    {
+        $assetUsages = $this->withAllRepositories(
+            function (ContentRepository $repository) use ($assetId) {
+                $filter = AssetUsageFilter::create()
+                    ->withAsset($assetId)
+                    ->includeVariantsOfAsset()
+                    ->groupByNode();
+
+                return $repository->projectionState(AssetUsageFinder::class)->findByFilter($filter);
+            }
+        );
+
+        return AssetUsages::fromArrayOfAssetUsages($assetUsages);
```

Comment:
I find this construct to make things less readable for no imminent gain (AFAICS).

Why not just:

```suggestion
        $filter = AssetUsageFilter::create()
            ->withAsset($assetId)
            ->includeVariantsOfAsset()
            ->groupByNode();
        $assetUsages = [];
        foreach ($this->getContentRepositories() as $contentRepository) {
            $assetUsages[] = $repository->projectionState(AssetUsageFinder::class)->findByFilter($filter);
        }
        return AssetUsages::fromArrayOfAssetUsages(array_merge(...$assetUsages));
```

Reply:
Sure, I'll replace it.